server/main_server/src/rodsServer.cpp

```diff
@@ -1184,26 +1208,32 @@ int serverMain(const bool enable_test_mode = false, const bool write_to_stdout =
             }
 
             if (irods::server_state::server_state::running != state) {
-                ix::log::server::info("Invalid iRODS server state [{}].", to_string(state));
+                log_server::info("Invalid iRODS server state [{}].", to_string(state));
             }
-
             FD_SET(svrComm.sock, &sockMask);
 
             int numSock = 0;
             struct timeval time_out;
             time_out.tv_sec  = 0;
             time_out.tv_usec = irods::SERVER_CONTROL_POLLING_TIME_MILLI_SEC * 1000;
+
             while ((numSock = select(svrComm.sock + 1, &sockMask, nullptr, nullptr, &time_out)) < 0) {
                 if (errno == EINTR) {
-                    rodsLog(LOG_NOTICE, "serverMain: select() interrupted");
+                    log_server::info("serverMain: select() interrupted");
                     FD_SET(svrComm.sock, &sockMask);
                     continue;
                 }
 
-                rodsLog(LOG_NOTICE, "serverMain: select() error, errno = %d", errno);
+                log_server::info("serverMain: select() error, errno = {}", errno);
                 return -1;
             }
 
+            if (reload_server_config) {
+                // As a side effect of reading the server properties object, we acquire a lock
+                // that cannot be had while the configuration is in the middle of reloading.
+                irods::server_properties::instance().contains("");
+            }
```

Comment:
What happens if we don't wait for the server to finish reloading?

Are there any risks?

Is it a bad idea to block until the server finishes reloading its config?

Reply:
> What happens if we don't wait for the server to finish reloading?

> Are there any risks?

There aren't any risks other than inconsistency, 

> Is it a bad idea to block until the server finishes reloading its config?

as to the blocking, well, how bad are we assuming disk latency is for our users? Is it high enough to make that a problem?